### PR TITLE
Add seqan2 to minimiser hash benchmark

### PR DESCRIPTION
**Pull request by MitraDarja**
Add seqan2 comparison to benchmark.
For ungapped seqan3 is sometimes slower, especially for k-mer=30. Strangely, the gapped version is supposed to be faster than seqan2 but if  I look at the kmer_hash_benchmark seqan2's kmer_hash is always faster than seqan3's.

<details> <summary> formatted benchmark </summary>

```
Method                      Throughput
seqan2_ungapped/1000/8/13   16.2115M/s
seqan2_ungapped/1000/8/18   17.1484M/s
seqan2_ungapped/1000/8/28   17.9737M/s
seqan2_ungapped/1000/30/35  13.348M/s
seqan2_ungapped/1000/30/40  13.779M/s
seqan2_ungapped/1000/30/50  14.4778M/s
seqan2_ungapped/50000/8/13  9.18865M/s
seqan2_ungapped/50000/8/18  10.523M/s
seqan2_ungapped/50000/8/28  10.4527M/s
seqan2_ungapped/50000/30/35 9.69901M/s
seqan2_ungapped/50000/30/40 10.0053M/s
seqan2_ungapped/50000/30/50 16.066M/s
seqan2_gapped/1000/8/13     12.6542M/s
seqan2_gapped/1000/8/18     12.9266M/s
seqan2_gapped/1000/8/28     13.6326M/s
seqan2_gapped/1000/30/35    8.56552M/s
seqan2_gapped/1000/30/40    9.87235M/s
seqan2_gapped/1000/30/50    10.1901M/s
seqan2_gapped/50000/8/13    9.35849M/s
seqan2_gapped/50000/8/18    8.98428M/s
seqan2_gapped/50000/8/28    13.9688M/s
seqan2_gapped/50000/30/35   9.53153M/s
seqan2_gapped/50000/30/40   10.2873M/s
seqan2_gapped/50000/30/50   9.84346M/s
naive/1000/8/13             11.2945M/s
naive/1000/8/18             6.21344M/s
naive/1000/8/28             4.88932M/s
naive/1000/30/35            5.07404M/s
naive/1000/30/40            4.74636M/s
naive/1000/30/50            4.42226M/s
naive/50000/8/13            7.89259M/s
naive/50000/8/18            5.38869M/s
naive/50000/8/28            5.3581M/s
naive/50000/30/35           5.74497M/s
naive/50000/30/40           5.68808M/s
naive/50000/30/50           3.8901M/s
seqan3_ungapped/1000/8/13   32.8807M/s
seqan3_ungapped/1000/8/18   32.179M/s
seqan3_ungapped/1000/8/28   34.1371M/s
seqan3_ungapped/1000/30/35  9.97492M/s
seqan3_ungapped/1000/30/40  10.3493M/s
seqan3_ungapped/1000/30/50  14.0404M/s
seqan3_ungapped/50000/8/13  21.4676M/s
seqan3_ungapped/50000/8/18  25.8773M/s
seqan3_ungapped/50000/8/28  27.5945M/s
seqan3_ungapped/50000/30/35 12.031M/s
seqan3_ungapped/50000/30/40 11.696M/s
seqan3_ungapped/50000/30/50 13.2557M/s
seqan3_gapped/1000/8/13     29.736M/s
seqan3_gapped/1000/8/18     36.1935M/s
seqan3_gapped/1000/8/28     31.0444M/s
seqan3_gapped/1000/30/35    12.5484M/s
seqan3_gapped/1000/30/40    11.3581M/s
seqan3_gapped/1000/30/50    11.622M/s
seqan3_gapped/50000/8/13    33.5967M/s
seqan3_gapped/50000/8/18    31.8625M/s
seqan3_gapped/50000/8/28    34.6677M/s
seqan3_gapped/50000/30/35   13.1168M/s
seqan3_gapped/50000/30/40   13.2588M/s
seqan3_gapped/50000/30/50   9.89449M/s
```

</details>
</details>

<details> <summary> full benchmark </summary>

```
compute_minimisers<method_tag::seqan2_ungapped>/1000/8/13        64182 ns        61253 ns         9898 Throughput[bp/s]=16.2115M/s
compute_minimisers<method_tag::seqan2_ungapped>/1000/8/18        67859 ns        57906 ns        12410 Throughput[bp/s]=17.1484M/s
compute_minimisers<method_tag::seqan2_ungapped>/1000/8/28        61558 ns        55248 ns        12241 Throughput[bp/s]=17.9737M/s
compute_minimisers<method_tag::seqan2_ungapped>/1000/30/35       83139 ns        72745 ns        12465 Throughput[bp/s]=13.348M/s
compute_minimisers<method_tag::seqan2_ungapped>/1000/30/40       88097 ns        70469 ns        10268 Throughput[bp/s]=13.779M/s
compute_minimisers<method_tag::seqan2_ungapped>/1000/30/50       76959 ns        67068 ns        10773 Throughput[bp/s]=14.4778M/s
compute_minimisers<method_tag::seqan2_ungapped>/50000/8/13     5816963 ns      5440737 ns          124 Throughput[bp/s]=9.18865M/s
compute_minimisers<method_tag::seqan2_ungapped>/50000/8/18     4854751 ns      4750839 ns          149 Throughput[bp/s]=10.523M/s
compute_minimisers<method_tag::seqan2_ungapped>/50000/8/28     4864279 ns      4782766 ns          127 Throughput[bp/s]=10.4527M/s
compute_minimisers<method_tag::seqan2_ungapped>/50000/30/35    5244650 ns      5152175 ns          137 Throughput[bp/s]=9.69901M/s
compute_minimisers<method_tag::seqan2_ungapped>/50000/30/40    5408603 ns      4994474 ns          114 Throughput[bp/s]=10.0053M/s
compute_minimisers<method_tag::seqan2_ungapped>/50000/30/50    3939195 ns      3110364 ns          209 Throughput[bp/s]=16.066M/s
compute_minimisers<method_tag::seqan2_gapped>/1000/8/13          78593 ns        78472 ns         7682 Throughput[bp/s]=12.6542M/s
compute_minimisers<method_tag::seqan2_gapped>/1000/8/18          77188 ns        76818 ns         7869 Throughput[bp/s]=12.9266M/s
compute_minimisers<method_tag::seqan2_gapped>/1000/8/28          73068 ns        72840 ns         9173 Throughput[bp/s]=13.6326M/s
compute_minimisers<method_tag::seqan2_gapped>/1000/30/35        118299 ns       113361 ns         5278 Throughput[bp/s]=8.56552M/s
compute_minimisers<method_tag::seqan2_gapped>/1000/30/40         98409 ns        98355 ns         6815 Throughput[bp/s]=9.87235M/s
compute_minimisers<method_tag::seqan2_gapped>/1000/30/50         95404 ns        95288 ns         6998 Throughput[bp/s]=10.1901M/s
compute_minimisers<method_tag::seqan2_gapped>/50000/8/13       5597388 ns      5341995 ns          100 Throughput[bp/s]=9.35849M/s
compute_minimisers<method_tag::seqan2_gapped>/50000/8/18       6054015 ns      5564498 ns          124 Throughput[bp/s]=8.98428M/s
compute_minimisers<method_tag::seqan2_gapped>/50000/8/28       3940929 ns      3578911 ns          190 Throughput[bp/s]=13.9688M/s
compute_minimisers<method_tag::seqan2_gapped>/50000/30/35      5464076 ns      5242707 ns          127 Throughput[bp/s]=9.53153M/s
compute_minimisers<method_tag::seqan2_gapped>/50000/30/40      6450332 ns      4857520 ns          152 Throughput[bp/s]=10.2873M/s
compute_minimisers<method_tag::seqan2_gapped>/50000/30/50      6260924 ns      5076569 ns          100 Throughput[bp/s]=9.84346M/s
compute_minimisers<method_tag::naive>/1000/8/13                  97696 ns        87919 ns         8814 Throughput[bp/s]=11.2945M/s
compute_minimisers<method_tag::naive>/1000/8/18                 183870 ns       159815 ns         4644 Throughput[bp/s]=6.21344M/s
compute_minimisers<method_tag::naive>/1000/8/28                 221210 ns       203096 ns         5069 Throughput[bp/s]=4.88932M/s
compute_minimisers<method_tag::naive>/1000/30/35                191398 ns       191366 ns         2949 Throughput[bp/s]=5.07404M/s
compute_minimisers<method_tag::naive>/1000/30/40                224089 ns       204578 ns         5139 Throughput[bp/s]=4.74636M/s
compute_minimisers<method_tag::naive>/1000/30/50                224936 ns       219571 ns         2731 Throughput[bp/s]=4.42226M/s
compute_minimisers<method_tag::naive>/50000/8/13               7509190 ns      6334172 ns          149 Throughput[bp/s]=7.89259M/s
compute_minimisers<method_tag::naive>/50000/8/18              10422725 ns      9277385 ns           77 Throughput[bp/s]=5.38869M/s
compute_minimisers<method_tag::naive>/50000/8/28              10284064 ns      9330365 ns           87 Throughput[bp/s]=5.3581M/s
compute_minimisers<method_tag::naive>/50000/30/35              9968479 ns      8698217 ns           69 Throughput[bp/s]=5.74497M/s
compute_minimisers<method_tag::naive>/50000/30/40             10845513 ns      8785217 ns          116 Throughput[bp/s]=5.68808M/s
compute_minimisers<method_tag::naive>/50000/30/50             13640315 ns     12845689 ns           75 Throughput[bp/s]=3.8901M/s
compute_minimisers<method_tag::seqan3_ungapped>/1000/8/13        37789 ns        30200 ns        20773 Throughput[bp/s]=32.8807M/s
compute_minimisers<method_tag::seqan3_ungapped>/1000/8/18        44794 ns        30859 ns        35909 Throughput[bp/s]=32.179M/s
compute_minimisers<method_tag::seqan3_ungapped>/1000/8/28        29883 ns        29089 ns        26456 Throughput[bp/s]=34.1371M/s
compute_minimisers<method_tag::seqan3_ungapped>/1000/30/35       98612 ns        97344 ns         6500 Throughput[bp/s]=9.97492M/s
compute_minimisers<method_tag::seqan3_ungapped>/1000/30/40      145620 ns        93823 ns         5577 Throughput[bp/s]=10.3493M/s
compute_minimisers<method_tag::seqan3_ungapped>/1000/30/50       70226 ns        69157 ns         9936 Throughput[bp/s]=14.0404M/s
compute_minimisers<method_tag::seqan3_ungapped>/50000/8/13     2798282 ns      2328768 ns          344 Throughput[bp/s]=21.4676M/s
compute_minimisers<method_tag::seqan3_ungapped>/50000/8/18     1984883 ns      1931926 ns          315 Throughput[bp/s]=25.8773M/s
compute_minimisers<method_tag::seqan3_ungapped>/50000/8/28     1855485 ns      1811701 ns          398 Throughput[bp/s]=27.5945M/s
compute_minimisers<method_tag::seqan3_ungapped>/50000/30/35    4356632 ns      4153533 ns          208 Throughput[bp/s]=12.031M/s
compute_minimisers<method_tag::seqan3_ungapped>/50000/30/40    4338191 ns      4272491 ns          175 Throughput[bp/s]=11.696M/s
compute_minimisers<method_tag::seqan3_ungapped>/50000/30/50    4045283 ns      3769764 ns          146 Throughput[bp/s]=13.2557M/s
compute_minimisers<method_tag::seqan3_gapped>/1000/8/13          38079 ns        33394 ns        18686 Throughput[bp/s]=29.736M/s
compute_minimisers<method_tag::seqan3_gapped>/1000/8/18          28120 ns        27436 ns        26240 Throughput[bp/s]=36.1935M/s
compute_minimisers<method_tag::seqan3_gapped>/1000/8/28          34230 ns        31986 ns        22193 Throughput[bp/s]=31.0444M/s
compute_minimisers<method_tag::seqan3_gapped>/1000/30/35         81658 ns        77380 ns        11607 Throughput[bp/s]=12.5484M/s
compute_minimisers<method_tag::seqan3_gapped>/1000/30/40         90668 ns        85490 ns         7562 Throughput[bp/s]=11.3581M/s
compute_minimisers<method_tag::seqan3_gapped>/1000/30/50         85232 ns        83549 ns         7906 Throughput[bp/s]=11.622M/s
compute_minimisers<method_tag::seqan3_gapped>/50000/8/13       1514479 ns      1488032 ns          420 Throughput[bp/s]=33.5967M/s
compute_minimisers<method_tag::seqan3_gapped>/50000/8/18       1615759 ns      1569024 ns          521 Throughput[bp/s]=31.8625M/s
compute_minimisers<method_tag::seqan3_gapped>/50000/8/28       1654738 ns      1442064 ns          452 Throughput[bp/s]=34.6677M/s
compute_minimisers<method_tag::seqan3_gapped>/50000/30/35      3871909 ns      3809708 ns          180 Throughput[bp/s]=13.1168M/s
compute_minimisers<method_tag::seqan3_gapped>/50000/30/40      4216049 ns      3768901 ns          186 Throughput[bp/s]=13.2588M/s
compute_minimisers<method_tag::seqan3_gapped>/50000/30/50      5164189 ns      5050389 ns          167 Throughput[bp/s]=9.89449M/s
```

</details>

### CODE REVIEW

File: test/performance/range/views/view_minimiser_hash_benchmark.cpp [hunk 1]
```diff
@@ -111,6 +116,86 @@ static void naive_minimiser_hash(benchmark::State & state)
...
+    seqan::String<seqan::Dna> seq_test = "ACGTCGACGTTTAG";
+    minimiser<use_xor::no> test;
+    seqan::Shape<seqan::Dna, seqan::SimpleShape> ss;
+    seqan::resize(ss, 4);
+    test.resize(window{8}, kmer{4});
+    test.compute(seq_test);
+    for (auto h : test.minimiser_hash)
+        std::cout << h<< "\n";*/
+
+    volatile size_t sum{0};
```
Comment (eseiler) [inline test/performance/range/views/view_minimiser_hash_benchmark.cpp:159]:
```suggestion
    size_t sum{0};
```

File: test/performance/range/views/view_minimiser_hash_benchmark.cpp [hunk 2]
```diff
@@ -111,6 +116,86 @@ static void naive_minimiser_hash(benchmark::State & state)
...
+    assert(sequence_length > 0);
+    size_t const k = static_cast<size_t>(state.range(1));
+    assert(k > 0);
+    size_t w = static_cast<size_t>(state.range(2));
+    assert(w > k);
+    auto seq = seqan3::test::generate_sequence_seqan2<seqan::Dna>(sequence_length, 0, 0);
+    minimiser<seqan::Shape<seqan::Dna, seqan::SimpleShape >> seqan_minimiser;
+    seqan::Shape<seqan::Dna, seqan::SimpleShape> s;
+    seqan::resize(s, k);
+    seqan_minimiser.resize(window{w}, kmer{k}, s);
```
Comment (eseiler) [inline test/performance/range/views/view_minimiser_hash_benchmark.cpp:145]:
```suggestion
    seqan::Shape<seqan::Dna, seqan::SimpleShape> simple_shape;
    seqan::resize(simple_shape, k);
    seqan_minimiser.resize(window{w}, kmer{k}, simple_shape);
```

File: test/performance/range/views/view_minimiser_hash_benchmark.cpp [hunk 3]
```diff
@@ -111,6 +116,86 @@ static void naive_minimiser_hash(benchmark::State & state)
...
+    auto sequence_length = state.range(0);
+    assert(sequence_length > 0);
+    size_t k = static_cast<size_t>(state.range(1));
+    assert(k > 0);
+    size_t w = static_cast<size_t>(state.range(2));
+    assert(w > k);
+    auto seq = seqan3::test::generate_sequence_seqan2<seqan::Dna>(sequence_length, 0, 0);
+    seqan::Shape<seqan::Dna, seqan::GenericShape> s = make_gapped_shape_seqan2(k);
+    minimiser<seqan::Shape<seqan::Dna, seqan::GenericShape>> seqan_minimiser;
+    seqan_minimiser.resize(window{w}, kmer{k}, s);
```
Comment (eseiler) [inline test/performance/range/views/view_minimiser_hash_benchmark.cpp:182]:
```suggestion
    seqan::Shape<seqan::Dna, seqan::GenericShape> generic_shape = make_gapped_shape_seqan2(k);
    minimiser<seqan::Shape<seqan::Dna, seqan::GenericShape>> seqan_minimiser;
    seqan_minimiser.resize(window{w}, kmer{k}, generic_shape);
```

File: test/include/seqan3/test/performance/seqan2_minimiser.h [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+// Copyright (c) 2006-2020, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2020, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+/*!\file
+ * \author Mitra Darvish <mitra.darvish AT fu-berlin.de>
+ * \brief Provides a SeqAn2 minimiser hash.
+ */
+#pragma once
```
Comment (eseiler) [inline test/include/seqan3/test/performance/seqan2_minimiser.h:11]:
```suggestion
// -----------------------------------------------------------------------------------------------------

/*!\file
 * \author Mitra Darvish <mitra.darvish AT fu-berlin.de>
 * \brief Provides a SeqAn2 minimiser hash.
 */
 
#pragma once
```

File: test/include/seqan3/test/performance/seqan2_minimiser.h [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \author Mitra Darvish <mitra.darvish AT fu-berlin.de>
+ * \brief Provides a SeqAn2 minimiser hash.
+ */
+ 
+#pragma once
+
+#ifdef SEQAN3_HAS_SEQAN2
```
Comment (eseiler) [inline test/include/seqan3/test/performance/seqan2_minimiser.h:15]:
```suggestion
#pragma once

#include <seqan3/core/platform.hpp>

#ifdef SEQAN3_HAS_SEQAN2
```

File: test/include/seqan3/test/performance/seqan2_minimiser.h [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+//!\brief Strong type for passing number of hash functions.
+struct hashes { uint64_t v; };
+
+//!\brief Whether to use xor for computing the hash value.
+enum use_xor : bool
+{
+    //!\brief Do not use xor.
+    no,
+    //!\brief Use xor.
+    yes
```
Comment (smehringer) [inline test/include/seqan3/test/performance/seqan2_minimiser.h:37]:
```suggestion
    no, //!< Do not use xor.
    yes //!< Use xor.
```
Those are equivalent. Just to show you a more compact version :)

File: test/include/seqan3/test/performance/seqan2_minimiser.h [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+struct kmer { uint64_t v; };
+//!\brief Strong type for passing number of bins.
+struct bins { uint64_t v; };
+//!\brief Strong type for passing number of bits.
+struct bits { uint64_t v; };
+//!\brief Strong type for passing number of hash functions.
+struct hashes { uint64_t v; };
+
+//!\brief Whether to use xor for computing the hash value.
+enum use_xor : bool
```
Comment (smehringer) [inline test/include/seqan3/test/performance/seqan2_minimiser.h:32]: Why do you have this enum if you do not change it in the benchmark? (You always use yes I think)

File: test/include/seqan3/test/performance/seqan2_minimiser.h [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+    shape_t reverse_shape{};
+
+    //!\brief Stores the k-mer hashes of the forward strand.
+    std::vector<uint64_t> forward_hashes;
+    //!\brief Stores the k-mer hashes of the reverse complement strand.
+    std::vector<uint64_t> reverse_hashes;
+
+
+public:
+
```
Comment (smehringer) [inline test/include/seqan3/test/performance/seqan2_minimiser.h:72]:
```suggestion

public:
```

File: test/include/seqan3/test/performance/seqan2_minimiser.h [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+     * \param[in] seed_ The new seed to use. Default: 0x8F3F73B5CF1C9ADE.
+     */
+     void resize(window const w_, kmer const k_, shape_t new_shape, uint64_t const seed_ = 0x8F3F73B5CF1C9ADE)
+   {
+       w = w_.v;
+       k = k_.v;
+       seed = seed_;
+       forward_shape = new_shape;
+       reverse_shape = new_shape;
+   }
```
Comment (smehringer) [inline test/include/seqan3/test/performance/seqan2_minimiser.h:111]:
```suggestion
    {
        w = w_.v;
        k = k_.v;
        seed = seed_;
        forward_shape = new_shape;
        reverse_shape = new_shape;
    }
```

File: test/include/seqan3/test/performance/seqan2_minimiser.h [hunk 10]
```diff
@@ -0,0 +1,221 @@
...
+    minimiser & operator=(minimiser const &) = default;           //!< Defaulted
+    minimiser & operator=(minimiser &&) = default;                //!< Defaulted
+    ~minimiser() = default;                                       //!< Defaulted
+
+    /*!\brief Constructs a minimiser from given k-mer, window size and a seed.
+     * \param[in] w_    The window size.
+     * \param[in] k_    The k-mer size.
+     * \param[in] seed_ The seed to use. Default: 0x8F3F73B5CF1C9ADE.
+     */
+    minimiser(window const w_, kmer const k_, uint64_t const seed_ = 0x8F3F73B5CF1C9ADE) :
```
Comment (smehringer) [inline test/include/seqan3/test/performance/seqan2_minimiser.h:92]: You don't use this constructor anywhere? Either remove the constructor or use it and remove the resize function.

File: test/include/seqan3/test/performance/seqan2_minimiser.h [hunk 11]
```diff
@@ -0,0 +1,221 @@
...
+            uint64_t reverse_hash = reverse_hashes[possible_kmers - i - 1];
+            window_values.emplace_back(std::min(forward_hash, reverse_hash), i, i + k - 1);
+        }
+
+        auto min = std::min_element(std::begin(window_values), std::end(window_values));
+        minimiser_hash.push_back(std::get<0>(*min));
+        minimiser_begin.push_back(std::get<1>(*min));
+        minimiser_end.push_back(std::get<2>(*min));
+
+
```
Comment (smehringer) [inline test/include/seqan3/test/performance/seqan2_minimiser.h:178]:
```suggestion

```

File: test/include/seqan3/test/performance/seqan2_minimiser.h [hunk 12]
```diff
@@ -0,0 +1,221 @@
...
+                minimiser_hash.push_back(std::get<0>(*min));
+                minimiser_begin.push_back(std::get<1>(*min));
+                minimiser_end.push_back(std::get<2>(*min));
+                minimiser_changed = false;
+            }
+        }
+        return;
+    }
+
+};
```
Comment (smehringer) [inline test/include/seqan3/test/performance/seqan2_minimiser.h:220]:
```suggestion
    }
};
```

**Comment by MitraDarja** [comment]
> Can you post the benchmark results here?

I edited my first post, now that I also have the gapped version, it looks better than in the beginning, but I am surprised that the gapped version is faster...

**Comment by eseiler** [comment]
SeqAn3 seems to be around twice as fast than SeqAn2; but the SeqAn2 version also looks at the revcomp...so we are probably at around the same speed once we have canonical minimizers?

**Comment by MitraDarja** [comment]
> Just some thoughts before I merge:
> 
> This is a quite big benchmark, How long does it take to execute?

/usr/bin/time says:
```
	User time (seconds): 54.50
	System time (seconds): 0.76
	Elapsed (wall clock) time (h:mm:ss or m:ss): 1:01.31
```

> Maybe we reduce the parameter scope a little bit? (You can always increase the scope manually)

Sure, any preference which parameters I should test or should not test?

> The times look weird. You posted the Throughput so I guess more is better.
> It seems that seqan3 is better when the small kmer sizes but worse for large kmer sizes? Or do I read this wrong?

That is how I would read it as well, but as Enrico pointed out currently, seqan3 does not support reverse, so thinking about the different behaviour makes more sense when the reverse implementation is included.

File: test/include/seqan3/test/performance/seqan2_minimiser.h [hunk 13]
```diff
@@ -4,16 +4,12 @@
...
-
 /*!\file
  * \author Mitra Darvish <mitra.darvish AT fu-berlin.de>
  * \brief Provides a SeqAn2 minimiser hash.
  */
- 
 #pragma once
 
-#include <seqan3/core/platform.hpp>
-
```
Comment (eseiler) [inline test/include/seqan3/test/performance/seqan2_minimiser.h]: You accidentally reverted the changes in these lines, hence the failing header test

**Comment by smehringer** [comment]
> > Just some thoughts before I merge:
> > This is a quite big benchmark, How long does it take to execute?
> 
> /usr/bin/time says:
> 
> ```
> 	User time (seconds): 54.50
> 	System time (seconds): 0.76
> 	Elapsed (wall clock) time (h:mm:ss or m:ss): 1:01.31
> ```
> 
> > Maybe we reduce the parameter scope a little bit? (You can always increase the scope manually)
> 
> Sure, any preference which parameters I should test or should not test?

A minute is quite long for CI. So I would remove the small sequence length (1000) and only have 2 instead of 3 different window sizes (remove the middle one). 

> > The times look weird. You posted the Throughput so I guess more is better.
> > It seems that seqan3 is better when the small kmer sizes but worse for large kmer sizes? Or do I read this wrong?
> 
> That is how I would read it as well, but as Enrico pointed out currently, seqan3 does not support reverse, so thinking about the different behaviour makes more sense when the reverse implementation is included.

That's true. We will try to evaluate and maybe profile the performance once the reverse minimizers are computed in seqan3

**Pull request merged**